File: app/src/prod/java/com/luigi573/mira/data/source/StatisticDataSourceModule.java
package com.luigi573.mira.data.source;

import com.luigi573.mira.data.source.local.StatisticsLocalDataSource;
import com.luigi573.mira.data.source.remote.StatisticsRemoteDataSource;

import javax.inject.Singleton;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class StatisticDataSourceModule {
    @Singleton
    @Binds
    @Local
    abstract StatisticsDataSource provideLocalDataSource(StatisticsLocalDataSource dataSource);

    @Singleton
    @Binds
    @Remote
    abstract StatisticsDataSource provideRemoteDataSource(StatisticsRemoteDataSource dataSource);
}

File: app/src/prod/java/com/luigi573/mira/data/source/StatisticDataSourceModule.kt
package com.luigi573.mira.data.source

import com.luigi573.mira.data.source.remote.StatRemoteDataSource
import com.luigi573.mira.di.AppModule
import javax.inject.Singleton
import dagger.Binds
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
object StatisticDataSourceModule {
    @JvmStatic
    @Singleton
    @AppModule.StatRemoteDataSource
    @Provides
    fun providesStatRemoteDataSource(retrofit: Retrofit) : StatDataSource {
        return StatRemoteDataSource(retrofit)
    }
}